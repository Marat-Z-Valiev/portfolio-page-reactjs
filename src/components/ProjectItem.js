import React from "react";
import Button from "./Button";
import "../css/Projects/ProjectItem/projectItem.css";

class ProjectItem extends React.Component {
	render() {
		const {
			name,
			description,
			tools,
			image,
			websiteLink,
			globeIcon,
			githubLink,
			githubIcon
		} = this.props.projects;
		return (
			<div>
				<hr className="line" />
				<h3>{name}</h3>
				<hr className="line" />
				<div className="project-description">
					<p>{`${name} ${description}`}</p>
					<p>
						<strong>{tools}</strong>
					</p>
				</div>
				<img className="portfolio-image img" src={image} alt={name} />
				<div className="buttons">
					<Button label="Website" link={websiteLink} icon={globeIcon} />
					<Button label="Github" link={githubLink} icon={githubIcon} />
				</div>
			</div>
		);
	}
}
export default ProjectItem;
