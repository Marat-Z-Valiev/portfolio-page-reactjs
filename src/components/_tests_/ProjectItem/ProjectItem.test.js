import React from "react";
import {shallow} from "enzyme";

import ProjectItem from "../../ProjectItem";

const component = shallow(
	<ProjectItem
		projects={{
			name: "Test",
			description: "Test",
			tools: "Test",
			image: "Test",
			websiteLink: "Test",
			globeIcon: "Test",
			githubLink: "Test",
			githubIcon: "Test"
		}}
	/>
);

describe("ProjectItem", () => {
	it("should render correctly", () => {
		expect(component).toMatchSnapshot();
	});
});
