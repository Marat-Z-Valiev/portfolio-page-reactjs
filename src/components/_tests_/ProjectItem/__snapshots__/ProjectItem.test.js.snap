// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectItem
    projects={
      Object {
        "description": "Test",
        "githubIcon": "Test",
        "githubLink": "Test",
        "globeIcon": "Test",
        "image": "Test",
        "name": "Test",
        "tools": "Test",
        "websiteLink": "Test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <hr
          className="line"
        />,
        <h3>
          Test
        </h3>,
        <hr
          className="line"
        />,
        <div
          className="project-description"
        >
          <p>
            Test Test
          </p>
          <p>
            <strong>
              Test
            </strong>
          </p>
        </div>,
        <img
          alt="Test"
          className="portfolio-image img"
          src="Test"
        />,
        <div
          className="buttons"
        >
          <Button
            icon="Test"
            label="Website"
            link="Test"
          />
          <Button
            icon="Test"
            label="Github"
            link="Test"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "line",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "line",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Test Test
            </p>,
            <p>
              <strong>
                Test
              </strong>
            </p>,
          ],
          "className": "project-description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test Test",
            },
            "ref": null,
            "rendered": "Test Test",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Test
              </strong>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
              },
              "ref": null,
              "rendered": "Test",
              "type": "strong",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Test",
          "className": "portfolio-image img",
          "src": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              icon="Test"
              label="Website"
              link="Test"
            />,
            <Button
              icon="Test"
              label="Github"
              link="Test"
            />,
          ],
          "className": "buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "icon": "Test",
              "label": "Website",
              "link": "Test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "icon": "Test",
              "label": "Github",
              "link": "Test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <hr
            className="line"
          />,
          <h3>
            Test
          </h3>,
          <hr
            className="line"
          />,
          <div
            className="project-description"
          >
            <p>
              Test Test
            </p>
            <p>
              <strong>
                Test
              </strong>
            </p>
          </div>,
          <img
            alt="Test"
            className="portfolio-image img"
            src="Test"
          />,
          <div
            className="buttons"
          >
            <Button
              icon="Test"
              label="Website"
              link="Test"
            />
            <Button
              icon="Test"
              label="Github"
              link="Test"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "line",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "line",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Test Test
              </p>,
              <p>
                <strong>
                  Test
                </strong>
              </p>,
            ],
            "className": "project-description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test Test",
              },
              "ref": null,
              "rendered": "Test Test",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Test
                </strong>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Test",
                },
                "ref": null,
                "rendered": "Test",
                "type": "strong",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Test",
            "className": "portfolio-image img",
            "src": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                icon="Test"
                label="Website"
                link="Test"
              />,
              <Button
                icon="Test"
                label="Github"
                link="Test"
              />,
            ],
            "className": "buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "icon": "Test",
                "label": "Website",
                "link": "Test",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "icon": "Test",
                "label": "Github",
                "link": "Test",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
